### ISSUE TITLE
Breaking NPOI

### ISSUE THREAD

**Issue by j4tech** [icsharpcode/SharpZipLib#270]
### Steps to reproduce
1. New C# project 
2. Install NPOI and then update SharpZipLib from 0.86.0 to 1.0.0
3. Create and save a simple excel file with NPOI

### Expected behavior
Excel file opens and is not corrupted.

### Actual behavior
Excel file is corrupted and does not open.

### Version of SharpZipLib
1.0.0

### Obtained from (place an x between the brackets for all that apply)
- [ ] Compiled from source
 - branch: _______
 - commit: _______
- [ ] Downloaded DLL from GitHub
- [ ] Downloaded DLL from SourceForge
- [ ] Downloaded DLL from _______
- [x ] DLL included as part of NPOI
- Package installed using:
 - [ x] NuGet
 - [ ] MyGet
 - [ ] Chocolatey

**Comment by piksel** [comment]
I have the following results from running tests:
https://gist.github.com/piksel/1cc46744f8a775483d26fa90c92d3b87

I cannot get NPOI v2.3 to produce any working output files at all. Not even with v0.86 of SharpZipLib.
Normally I would suggest opening an issue at [tonyqus/npoi](https://github.com/tonyqus/npoi) but he has got issues disabled.

[DotNetCore.NPOI](https://github.com/dotnetcore/NPOI) seems to work fine though, perhaps it could be used instead?

**Comment by markhobson** [comment]
If it helps, this is the error that I get when using NPOI 2.3.0 with SharpZipLib 1.0.0:
```
System.MethodAccessException
  Message=Attempt by security transparent method 'NPOI.OpenXml4Net.OPC.ZipPackage..ctor(System.IO.Stream, NPOI.OpenXml4Net.OPC.PackageAccess)' to access security critical method 'ICSharpCode.SharpZipLib.Zip.ZipInputStream..ctor(System.IO.Stream)' failed.

Assembly 'NPOI.OpenXml4Net, Version=2.3.0.0, Culture=neutral, PublicKeyToken=0df73ec7942b34e1' is marked with the AllowPartiallyTrustedCallersAttribute, and uses the level 2 security transparency model.  Level 2 transparency causes all methods in AllowPartiallyTrustedCallers assemblies to become security transparent by default, which may be the cause of this exception.
```

**Comment by snowfire20** [comment]
please see https://stackoverflow.com/questions/52432909/npoi-writes-0-bytes-in-the-memorystream/52758332#52758332
https://stackoverflow.com/questions/52725550/npoi-version-2-2-memorystream-toarray-0-bytes

NPOI Version 2.2 and 2.3 do not work with SharpZipLib 1.0.0. the current workaround is to downgrade to version .86 which works.

**Comment by piksel** [comment]
Can someone make a small repro example that experiences the problem? The example code I found did not work with 0.86 either. Or if you see something wrong with my gist, just make a new revision: https://gist.github.com/piksel/1cc46744f8a775483d26fa90c92d3b87

**Comment by tonyqus** [comment]
Hi, I'm not sure what's going on. But NPOI 2.3.0 only works on SharpZipLib 0.86 for now.  I'm working on NPOI 2.4.0 version these days which will be built based on SharpZipLib 1.0.0. According to the test cases execution, I don't see SharpZipLib 1.0.0 as a blocker.

Btw, this issue should be opened in NPOI issue list instead of SharpZipLib. SharpZipLib did a good job as a basis library. I really appreciate that.

**Comment by tonyqus** [comment]
@markhobson  As I know, this error is caused by the new security transparent model introduced by .NET 4.x.  Please read https://docs.microsoft.com/en-us/dotnet/framework/misc/security-transparent-code-level-1. Microsoft is getting rid of old assembly attribute AllowPartiallyTrustedCallersAttribute, which is used in NPOI assembly setting.

**Comment by piksel** [comment]
I am closing this as invalid since it's not something we can solve in the library.

But as a summary: **until v2.4 of NPOI is released, use version 0.86 of SharpZipLib**.
